org-roam.el

```diff
@@ -451,20 +451,54 @@ INFO is an alist containing additional information."
 ;;;; org-roam-link-face
 (defface org-roam-link
   '((t :inherit org-link))
-  "Face for org-roam link."
+  "Face for Org-roam links."
   :group 'org-roam-faces)
 
-(defface org-roam-backlink
+(defface org-roam-link-current
   '((t :inherit org-block))
-  "Face for org-roam backlinks in backlinks buffer"
+  "Face for Org-roam links pointing to the current buffer."
   :group 'org-roam-faces)
 
+(defun org-roam--in-buffer-p ()
+  "Return t if in the Org-roam buffer."
+  (string= (buffer-name)
+           org-roam-buffer))
+
+(defun org-roam--retrieve-link-path (&optional pom)
+  "Retrieve the path of the link at point or POM.
+POM can be a position in the current buffer or a marker."
+  (let ((pom (or pom (point))))
+    (org-with-point-at pom
+      (plist-get (cadr (org-element-context)) :path))))
+
+(defun org-roam--backlink-to-current-p ()
+  "Return t if backlink is to the current Org-roam file."
+  (let* ((current (buffer-file-name org-roam-buffer--current))
+         (backlink-dest (org-roam--retrieve-link-path))
+         ;; Retrieve path from file-heading
+         (origin-file (save-excursion
+                        (org-back-to-heading)
+                        (re-search-forward "^\\*+ " nil t)
+                        (org-roam--retrieve-link-path)))
+         ;; Compute absolute path to `backlink-dest'
+         (backlink-dest-abs (relative-expand-file-name
```

Comment:
Where's `relative-expand-file-name` comes from? This causes `Error during redisplay: (jit-lock-function 1) signaled (void-function relative-expand-file-name)` in my Emacs.

Reply:
Damn, you're right.  I think it was native, but it's from `load-relative`.  I took me quite a while to find a function that did what I wanted, i.e. to compute an absolute path from 1) another absolute path, and 2) a relative path from that 2nd absolute path.

I'll see what I can do.

Reply:
It's easy to refactor.  I'll take care of it now.

Reply:
Addressed in 85975de650d8dd13b781d3b8553efd654fcac560.